<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Provides access to smokingMouse project data • smokingMouse</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Provides access to smokingMouse project data">
<meta property="og:description" content="This is an ExperimentHub package that provides access to the data at the gene, exon, transcript and junction level used in the analyses of the smokingMouse project. See https://github.com/LieberInstitute/smokingMouse_Indirects. This datasets contain the expression counts of genes, transcripts, exons and exon-exon junctions across 208 mice samples from pup and adult brains and adult blood. They also contain relevant information of these samples and features, such as conditions, QC metrics and if they were used after filtering steps and also if the features were differently expressed in the different experiments.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">smokingMouse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/smokingMouse.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/smokingMouse/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="smokingmouse">smokingMouse<a class="anchor" aria-label="anchor" href="#smokingmouse"></a>
</h1></div>
<!-- badges: start -->

<!-- badges: end -->
<p>Welcome to the <code>smokingMouse</code> project! Here you are able to access the mouse and human datasets used for the analysis of the smoking mouse LIBD project.</p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This bulk RNA-sequencing project consisted of a differential expression analysis (DEA) involving 4 data types: genes, exons, transcripts and junctions. The main goal of this study was to explore the effects of prenatal exposure to maternal smoking and nicotine exposures on the developing mouse brain. As secondary objectives, this work evaluated: 1) the affected genes by each exposure on the adult female brain in order to compare offspring and adult results and 2) the effects of smoking on adult blood and brain to search for overlapping biomarkers in both tissues. Finally, DEGs identified in mice were compared against previously published results from a human study (Semick, S.A. et al. (2018)).</p>
<p>The next table summarizes the analyses done at each level.</p>
<figure><img src="reference/figures/Table_of_Analyses.png" align="center" width="800px"><figcaption style="color: gray; line-height: 0.94; text-align: justify"><p><font size="-1.5"><b>Table 1</b>: <b>1. Data preparation</b>: in this first step, counts of genes, exons and junctions were normalized to CPM and scaled; transcript expression values were only scaled since they were already in TMP. Then, low-expression features were removed using the indicated methods and samples were separated by tissue and age in order to create subsets of the data for downstream analyses. <b>2. Exploratory Data Analysis</b>: QC metrics of the samples were examined and used to filter them; sample level effects were explored through dimensionality reduction methods and rare samples in PCA plots were manually removed from the datasets; gene level effects were evaluated with analyses of explanatory variables and variance partition. <b>3. Differential Expression Analysis</b>: with the relevant variables identified in the previous steps, the DEA was performed at the gene level for nicotine and smoking, adult and pup, and blood and brain samples, and for 3 models: the naive one modeled <span style="font-family: monospace">~Group + batch effects</span>, the adjusted model modeled <span style="font-family: monospace">~Group + Pregnancy + batch effects</span> for adults and <span style="font-family: monospace">~Group + Sex + batch effects</span> for pups, and the interaction model <span style="font-family: monospace">~Group*Pregnancy + batch effects</span> for adults and <span style="font-family: monospace">~Group*Sex + batch effects</span> for pups; DEA on the rest of the levels was performed for pups only and using the adjusted model. After that, signals of the features in nicotine and smoking were compared, as well as the signals of exons and txs vs the effects of their genes, and genes’ signals were additionally compared in the different tissues, ages, models and species (vs human data of a previous study). All resultant DEG and DE features (and their genes) were quantified and compared based on their experiment (nic/smo) and direction of regulation (up/down); DEG were further compared against genes of DE exons and txs; mouse genes were also compared with human genes affected by cigarette smoke or associated with TUD. <b>4. Gene Ontology and KEGG</b>: taking the DEG and the genes of DE txs and exons, GO &amp; KEGG analyses were done and the expression levels of genes that participate in brain development related processes were explored. <b>5. DE feature visualization</b>: DEG counts were represented in heatmaps in order to distinguish the groups of up and down-regulated genes. <b>6. Junction annotation</b>: for novel DE jxns of unknown gene, their nearest, preceding and following genes were determined.</font></p>
<p><font size="0.8">Abbreviations: Jxn: junction; Tx: transcript; CPM: counts per million; TPM: transcripts per million; TMM: Trimmed Mean of M-Values; TMMwsp: TMM with singleton pairing; EDA: exploratory data analysis; QC: quality control; ribo: ribosomal; mt: mitochondrial; PCA: Principal Component Analysis; PC: principal component; MDS: Multidimensional Scaling; DEA: differential expression analysis; DE: differential expression/differentially expressed; DEG: differentially expressed genes; GO: Gene Ontology; KEGG: Kyoto Encyclopedia of Genes and Genomes; TUD: tobacco use disorder.</font></p>
</figcaption></figure>
</div>
<div class="section level2">
<h2 id="study-design">Study design<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h2>
<figure><img src="reference/figures/Study_design.png" align="center" width="800px"><figcaption style="color: gray; line-height: 0.88; text-align: justify"><p><font size="-1.5"><b>Figure 1</b>: <b>Experimental design of the study</b>. <b>A)</b> 36 pregnant dams and 35 non-pregnant female adult mice were either administered nicotine by intraperitoneal injection (IP; n=12), exposed to cigarette smoke in smoking chambers (n=24), or controls (n=35; 11 nicotine controls and 24 smoking controls). A total of 137 pups were born to pregnant dams: 19 were born to mice that were administered nicotine, 46 to mice exposed to cigarette smoke and the remaining 72 to control mice (23 to nicotine controls and 49 to smoking controls). Samples from frontal cortices of P0 pups and adults were obtained, as well as blood samples from smoking-exposed and smoking control adults. <b>B)</b> RNA was extracted, RNA-seq libraries were prepared and sequenced to obtain expression counts for genes, exons, transcripts and exon-exon junctions.</font></p>
</figcaption></figure>
</div>
<div class="section level2">
<h2 id="smoking-mouse-datasets">smoking Mouse datasets<a class="anchor" aria-label="anchor" href="#smoking-mouse-datasets"></a>
</h2>
<p>The mouse datasets contain the following data in a single object for each feature (genes, exons, transcripts and exon-exon junctions):</p>
<ul>
<li>
<strong>Raw data</strong>: original read counts of the features, also including the original information of features and samples.</li>
<li>
<strong>Processed data</strong>: normalized and scaled counts of the same features. In addition to the feature and sample information, the datasets contain information of which ones were used in downstream analyses (the ones that passed filtering steps) and which features were DE in the different experiments.</li>
</ul>
<p>Moreover, you can find human data generated by Semick, S.A. et al. (2018) in Mol Psychiatry, DOI: <a href="https://doi.org/10.1038/s41380-018-0223-1" class="external-link uri">https://doi.org/10.1038/s41380-018-0223-1</a>, that contain the results of a DEA in adult and prenatal human brain samples exposed to cigarette smoke.</p>
</div>
<div class="section level2">
<h2 id="data-specifics">Data specifics<a class="anchor" aria-label="anchor" href="#data-specifics"></a>
</h2>
<ul>
<li>
<em>‘rse_gene_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_gene object) the gene RSE object contains raw and normalized expression data of 55401 mouse genes across 208 samples from brains and blood of healthy and nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_tx_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_tx object) the tx RSE object contains raw and normalized expression data of 142604 mouse transcripts across 208 samples from brains and blood of healthy and nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_exon_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_exon object) the exon RSE object contains raw and normalized expression data of 447670 mouse exons across 208 samples from brains and blood of healthy and nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_jx_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_jx object) the jx RSE object contains raw and normalized expression data of 1436068 mouse exon-exon junctions across 208 samples from brains and blood of healthy and nicotine/smoking-exposed pup and adult mice.</li>
</ul>
<p>All the above datasets contain sample and feature information and additional data of the results obtained in the filtering steps and the DEA.</p>
<ul>
<li>
<em>‘de_genes_prenatal_human_brain_smoking.Rdata’</em>: (object with the same name) data frame with DE (ctrls vs smoking-exposed samples) data of 18067 genes in human prenatal brain samples exposed to cigarette smoke.</li>
<li>
<em>‘de_genes_adult_human_brain_smoking.Rdata’</em>: (object with the same name) data frame with DE (ctrls vs smoking-exposed samples) data of 18067 genes in human adult brain samples exposed to cigarette smoke.</li>
</ul>
</div>
<div class="section level2">
<h2 id="rbioconductor-package">R/Bioconductor package<a class="anchor" aria-label="anchor" href="#rbioconductor-package"></a>
</h2>
<p>The <code>smokingMouse</code> package contains functions for:</p>
<ul>
<li>Accessing the expression data from the LIBD smoking Mouse project (<a href="https://github.com/LieberInstitute/smokingMouse_Indirects" class="external-link">code on GitHub</a>). The datasets are retrieved from <a href="http://bioconductor.org/" class="external-link">Bioconductor</a>’s <code>ExperimentHub</code>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/" class="external-link">CRAN</a>. Then install <code>smokingMouse</code> from <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> using the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"smokingMouse"</span><span class="op">)</span></span></code></pre></div>
<p>And the development version from <a href="https://github.com/LieberInstitute/smokingMouse" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"LieberInstitute/smokingMouse"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-of-how-to-access-the-data">Example of how to access the data<a class="anchor" aria-label="anchor" href="#example-of-how-to-access-the-data"></a>
</h2>
<p>Through the <code>smokingMouse</code> package you can access the mouse datasets of the project that include the raw and processed data. Below there’s code you can use to access the gene data but can do the same for any of the datasets. For more details, check the documentation for <code>RangedSummarizedExperiment</code> objects. You can also find code to access human data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Connect to ExperimentHub</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'ExperimentHub' was built under R version 4.3.1</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationHub</span></span>
<span><span class="co">#&gt; Loading required package: BiocFileCache</span></span>
<span><span class="co">#&gt; Warning: package 'BiocFileCache' was built under R version 4.3.1</span></span>
<span><span class="co">#&gt; Loading required package: dbplyr</span></span>
<span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Cannot connect to ExperimentHub server, using 'localHub=TRUE' instead</span></span>
<span><span class="co">#&gt; Using 'localHub=TRUE'</span></span>
<span><span class="co">#&gt;   If offline, please also see BiocManager vignette section on offline use</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> </span>
<span><span class="co">## Load the datasets of the package</span></span>
<span><span class="va">myfiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="st">"smokingMouse"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="co">########################</span></span>
<span><span class="co">#      Mouse data </span></span>
<span><span class="co">########################</span></span>
<span><span class="co">## Download the mouse gene data</span></span>
<span><span class="va">rse_gene</span> <span class="op">&lt;-</span> <span class="va">myfiles</span><span class="op">[[</span><span class="st">'EH8313'</span><span class="op">]</span><span class="op">]</span> </span>
<span><span class="co">## This is a RangedSummarizedExperiment object</span></span>
<span><span class="va">rse_gene</span></span>
<span></span>
<span><span class="co">## Check sample info </span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">## Check gene info</span></span>
<span><span class="fu">rowData</span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">## Access the original counts</span></span>
<span><span class="va">original_counts</span> <span class="op">&lt;-</span> <span class="fu">assays</span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="co">## Access the log normalized counts</span></span>
<span><span class="va">logcounts</span> <span class="op">&lt;-</span> <span class="fu">assays</span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span></span>
<span></span>
<span></span>
<span><span class="co">########################</span></span>
<span><span class="co">#      Human data </span></span>
<span><span class="co">########################</span></span>
<span><span class="co">## Download the human gene data</span></span>
<span><span class="va">de_genes_prenatal_human_brain_smoking</span> <span class="op">&lt;-</span> <span class="va">myfiles</span><span class="op">[[</span><span class="st">'EH8317'</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">## This is a data frame</span></span>
<span><span class="va">de_genes_prenatal_human_brain_smoking</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Access data of human genes as normally do with data frames</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Below is the citation output from using <code>citation('smokingMouse')</code> in R. Please run this yourself to check for any updates on how to cite <strong>smokingMouse</strong>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">'smokingMouse'</span><span class="op">)</span>, bibtex <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'smokingMouse' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Gonzalez-Padilla D, Collado-Torres L (2023). _Provides access to</span></span>
<span><span class="co">#&gt;   smokingMouse project data _. doi:10.18129/B9.bioc.smokingMouse</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.18129/B9.bioc.smokingMouse&gt;,</span></span>
<span><span class="co">#&gt;   https://github.com/LieberInstitute/smokingMouse/smokingMouse - R</span></span>
<span><span class="co">#&gt;   package version 0.99.5,</span></span>
<span><span class="co">#&gt;   &lt;http://www.bioconductor.org/packages/smokingMouse&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {Provides access to smokingMouse project data },</span></span>
<span><span class="co">#&gt;     author = {Daianna Gonzalez-Padilla and Leonardo Collado-Torres},</span></span>
<span><span class="co">#&gt;     year = {2023},</span></span>
<span><span class="co">#&gt;     url = {http://www.bioconductor.org/packages/smokingMouse},</span></span>
<span><span class="co">#&gt;     note = {https://github.com/LieberInstitute/smokingMouse/smokingMouse - R package version 0.99.5},</span></span>
<span><span class="co">#&gt;     doi = {10.18129/B9.bioc.smokingMouse},</span></span>
<span><span class="co">#&gt;   }</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Gonzalez-Padilla D, Collado-Torres L (2023). "Provides access to</span></span>
<span><span class="co">#&gt;   smokingMouse project data." _bioRxiv_. doi:10.1101/TODO</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/TODO&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://www.biorxiv.org/content/10.1101/TODO&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {Provides access to smokingMouse project data},</span></span>
<span><span class="co">#&gt;     author = {Daianna Gonzalez-Padilla and Leonardo Collado-Torres},</span></span>
<span><span class="co">#&gt;     year = {2023},</span></span>
<span><span class="co">#&gt;     journal = {bioRxiv},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/TODO},</span></span>
<span><span class="co">#&gt;     url = {https://www.biorxiv.org/content/10.1101/TODO},</span></span>
<span><span class="co">#&gt;   }</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To cite the original work from which human data come please use the following citation:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Semick, S. A., Collado-Torres, L., Markunas, C. A., Shin, J. H., Deep-Soboslay, A., Tao, R., ... </span></span>
<span><span class="co">#&gt; &amp; Jaffe, A. E. (2020). Developmental effects of maternal smoking during pregnancy on the human</span></span>
<span><span class="co">#&gt;  frontal cortex transcriptome. Molecular psychiatry, 25(12), 3267-3277.</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>Please note that the <code>smokingMouse</code> and the <a href="https://github.com/LieberInstitute/smokingMouse_Indirects" class="external-link">smoking mouse</a> project were only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the <code>smokingMouse</code> project is released with a <a href="http://bioconductor.org/about/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
<div class="section level2">
<h2 id="development-tools">Development tools<a class="anchor" aria-label="anchor" href="#development-tools"></a>
</h2>
<ul>
<li>Continuous code testing is possible thanks to <a href="https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/" class="external-link">GitHub actions</a> through <em><a href="https://CRAN.R-project.org/package=usethis" class="external-link">usethis</a></em>, <em><a href="https://CRAN.R-project.org/package=remotes" class="external-link">remotes</a></em>, and <em><a href="https://CRAN.R-project.org/package=rcmdcheck" class="external-link">rcmdcheck</a></em> customized to use <a href="https://www.bioconductor.org/help/docker/" class="external-link">Bioconductor’s docker containers</a> and <em><a href="https://bioconductor.org/packages/3.18/BiocCheck" class="external-link">BiocCheck</a></em>.</li>
<li>Code coverage assessment is possible thanks to <a href="https://codecov.io/gh" class="external-link">codecov</a> and <em><a href="https://CRAN.R-project.org/package=covr" class="external-link">covr</a></em>.</li>
<li>The <a href="http://LieberInstitute.github.io/smokingMouse" class="external-link">documentation website</a> is automatically updated thanks to <em><a href="https://CRAN.R-project.org/package=pkgdown" class="external-link">pkgdown</a></em>.</li>
<li>The code is styled automatically thanks to <em><a href="https://CRAN.R-project.org/package=styler" class="external-link">styler</a></em>.</li>
<li>The documentation is formatted thanks to <em><a href="https://CRAN.R-project.org/package=devtools" class="external-link">devtools</a></em> and <em><a href="https://CRAN.R-project.org/package=roxygen2" class="external-link">roxygen2</a></em>.</li>
</ul>
<p>For more details, check the <code>dev</code> directory.</p>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.18/biocthis" class="external-link">biocthis</a></em>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/LieberInstitute/smokingMouse/" class="external-link">Browse source code</a></li>
<li><a href="https://support.bioconductor.org/t/smokingMouse" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
<li><a href="SUPPORT.html">Getting help</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing smokingMouse</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Daianna Gonzalez-Padilla <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0009-0005-8348-3195" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/LieberInstitute/smokingMouse/issues" class="external-link"><img src="https://img.shields.io/github/issues/LieberInstitute/smokingMouse" alt="GitHub issues"></a></li>
<li><a href="https://github.com/LieberInstitute/smokingMouse/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/LieberInstitute/smokingMouse" alt="GitHub pulls"></a></li>
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://bioconductor.org/checkResults/release/data-experiment-LATEST/smokingMouse" class="external-link"><img src="http://www.bioconductor.org/shields/build/release/data-experiment/smokingMouse.svg" alt="Bioc release status"></a></li>
<li><a href="https://bioconductor.org/checkResults/devel/data-experiment-LATEST/smokingMouse" class="external-link"><img src="http://www.bioconductor.org/shields/build/devel/data-experiment/smokingMouse.svg" alt="Bioc devel status"></a></li>
<li><a href="http://bioconductor.org/packages/stats/bioc/smokingMouse/" class="external-link"><img src="https://bioconductor.org/shields/downloads/release/smokingMouse.svg" alt="Bioc downloads rank"></a></li>
<li><a href="https://support.bioconductor.org/tag/smokingMouse" class="external-link"><img src="https://bioconductor.org/shields/posts/smokingMouse.svg" alt="Bioc support"></a></li>
<li><a href="http://bioconductor.org/checkResults/devel/data-experiment-LATEST/smokingMouse/" class="external-link"><img src="https://bioconductor.org/shields/lastcommit/devel/data-experiment/smokingMouse.svg" alt="Bioc last commit"></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/smokingMouse.html#since" class="external-link"><img src="https://bioconductor.org/shields/dependencies/release/smokingMouse.svg" alt="Bioc dependencies"></a></li>
<li><a href="https://github.com/LieberInstitute/smokingMouse/actions/workflows/R-CMD-check-bioc.yaml" class="external-link"><img src="https://github.com/LieberInstitute/smokingMouse/actions/workflows/R-CMD-check-bioc.yaml/badge.svg" alt="R-CMD-check-bioc"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Daianna Gonzalez-Padilla.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
