<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to smokingMouse • smokingMouse</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Introduction to smokingMouse">
<meta property="og:description" content="smokingMouse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">smokingMouse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/smokingMouse.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/smokingMouse/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to smokingMouse</h1>
                        <h4 data-toc-skip class="author">Daianna
Gonzalez-Padilla</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development
(LIBD)<br><a class="author_email" href="mailto:#"></a><a href="mailto:glezdaianna@gmail.com" class="email">glezdaianna@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">12 July 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/smokingMouse/blob/HEAD/vignettes/smokingMouse.Rmd" class="external-link"><code>vignettes/smokingMouse.Rmd</code></a></small>
      <div class="hidden name"><code>smokingMouse.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Welcome to the <code>smokingMouse</code> project. In this vignette
we’ll show you how to access the smokingMouse LIBD datasets . You can
find the analysis code and the data generation in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects" class="external-link">here</a>.</p>
<div class="section level3">
<h3 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h3>
<p>The main motivation to create this bioconductor package was to
provide public and free access to all RNA-seq datasets that were
generated for the smokingMouse project, containing many variables of
interest that make it possible to answer a wide range of biological
questions related to smoking and nicotine effects in mice.</p>
</div>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>This bulk RNA-sequencing project consisted of a differential
expression analysis (DEA) involving 4 data types: genes, exons,
transcripts and junctions. The main goal of this study was to explore
the effects of prenatal exposure to maternal smoking and nicotine
exposures on the developing mouse brain. As secondary objectives, this
work evaluated: 1) the affected genes by each exposure on the adult
female brain in order to compare offspring and adult results and 2) the
effects of smoking on adult blood and brain to search for overlapping
biomarkers in both tissues. Finally, DEGs identified in mice were
compared against previously published results from a human study
(Semick, S.A. et al. (2018)).</p>
<p>The next table summarizes the analyses done at each level.</p>
<figure><img src="http://research.libd.org/smokingMouse/reference/figures/Table_of_Analyses.png" align="center" width="800px"><figcaption style="color: gray; line-height: 0.94; text-align: justify"><p><font size="-1.5"><b>Table 1</b>: <b>1. Data preparation</b>: in this
first step, counts of genes, exons and junctions were normalized to CPM
and scaled; transcript expression values were only scaled since they
were already in TMP. Then, low-expression features were removed using
the indicated methods and samples were separated by tissue and age in
order to create subsets of the data for downstream analyses. <b>2.
Exploratory Data Analysis</b>: QC metrics of the samples were examined
and used to filter them; sample level effects were explored through
dimensionality reduction methods and rare samples in PCA plots were
manually removed from the datasets; gene level effects were evaluated
with analyses of explanatory variables and variance partition. <b>3.
Differential Expression Analysis</b>: with the relevant variables
identified in the previous steps, the DEA was performed at the gene
level for nicotine and smoking, adult and pup, and blood and brain
samples, and for 3 models: the naive one modeled <span style="font-family: monospace">~Group + batch effects</span>, the
adjusted model modeled <span style="font-family: monospace">~Group +
Pregnancy + batch effects</span> for adults and <span style="font-family: monospace">~Group + Sex + batch effects</span> for
pups, and the interaction model <span style="font-family: monospace">~Group*Pregnancy + batch effects</span>
for adults and <span style="font-family: monospace">~Group*Sex + batch
effects</span> for pups; DEA on the rest of the levels was performed for
pups only and using the adjusted model. After that, signals of the
features in nicotine and smoking were compared, as well as the signals
of exons and txs vs the effects of their genes, and genes’ signals were
additionally compared in the different tissues, ages, models and species
(vs human data of a previous study). All resultant DEG and DE features
(and their genes) were quantified and compared based on their experiment
(nic/smo) and direction of regulation (up/down); DEG were further
compared against genes of DE exons and txs; mouse genes were also
compared with human genes affected by cigarette smoke or associated with
TUD. <b>4. Gene Ontology and KEGG</b>: taking the DEG and the genes of
DE txs and exons, GO &amp; KEGG analyses were done and the expression
levels of genes that participate in brain development related processes
were explored. <b>5. DE feature visualization</b>: DEG counts were
represented in heatmaps in order to distinguish the groups of up and
down-regulated genes. <b>6. Junction annotation</b>: for novel DE jxns
of unknown gene, their nearest, preceding and following genes were
determined.</font></p>
<p><font size="0.8">Abbreviations: Jxn: junction; Tx: transcript; CPM:
counts per million; TPM: transcripts per million; TMM: Trimmed Mean of
M-Values; TMMwsp: TMM with singleton pairing; EDA: exploratory data
analysis; QC: quality control; ribo: ribosomal; mt: mitochondrial; PCA:
Principal Component Analysis; PC: principal component; MDS:
Multidimensional Scaling; DEA: differential expression analysis; DE:
differential expression/differentially expressed; DEG: differentially
expressed genes; GO: Gene Ontology; KEGG: Kyoto Encyclopedia of Genes
and Genomes; TUD: tobacco use disorder.</font></p>
</figcaption></figure>
</div>
<div class="section level3">
<h3 id="study-design">Study design<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h3>
<figure><img src="http://research.libd.org/smokingMouse/reference/figures/Study_design.png" align="center" width="800px"><figcaption style="color: gray; line-height: 0.88; text-align: justify"><p><font size="-1.5"><b>Figure 1</b>: <b>Experimental design of the
study</b>. <b>A)</b> 36 pregnant dams and 35 non-pregnant female adult
mice were either administered nicotine by intraperitoneal injection (IP;
n=12), exposed to cigarette smoke in smoking chambers (n=24), or
controls (n=35; 11 nicotine controls and 24 smoking controls). A total
of 137 pups were born to pregnant dams: 19 were born to mice that were
administered nicotine, 46 to mice exposed to cigarette smoke and the
remaining 72 to control mice (23 to nicotine controls and 49 to smoking
controls). Samples from frontal cortices of P0 pups and adults were
obtained, as well as blood samples from smoking-exposed and smoking
control adults. <b>B)</b> RNA was extracted, RNA-seq libraries were
prepared and sequenced to obtain expression counts for genes, exons,
transcripts and exon-exon junctions.</font></p>
</figcaption></figure>
</div>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-smokingmouse">Install <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#install-smokingmouse"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
is a <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages.
<code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"smokingMouse"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
is based on many other packages and in particular in those that have
implemented the infrastructure needed for dealing with RNA-seq data and
differential expression. That is, packages like <em><a href="https://bioconductor.org/packages/3.17/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
and <em><a href="https://bioconductor.org/packages/3.17/limma" class="external-link">limma</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help: remember to use the
<code>smokingMouse</code> tag and check <a href="https://support.bioconductor.org/tag/smokingMouse/" class="external-link">the older
posts</a>. Other alternatives are available such as creating GitHub
issues and tweeting. However, please note that if you want to receive
help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="citing-smokingmouse">Citing <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#citing-smokingmouse"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"smokingMouse"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'smokingMouse' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Gonzalez-Padilla D, Collado-Torres L (2023). _Provides access to</span></span>
<span><span class="co">#&gt;   smokingMouse project data _. doi:10.18129/B9.bioc.smokingMouse</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.18129/B9.bioc.smokingMouse&gt;,</span></span>
<span><span class="co">#&gt;   https://github.com/LieberInstitute/smokingMouse/smokingMouse - R</span></span>
<span><span class="co">#&gt;   package version 0.99.5,</span></span>
<span><span class="co">#&gt;   &lt;http://www.bioconductor.org/packages/smokingMouse&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {Provides access to smokingMouse project data },</span></span>
<span><span class="co">#&gt;     author = {Daianna Gonzalez-Padilla and Leonardo Collado-Torres},</span></span>
<span><span class="co">#&gt;     year = {2023},</span></span>
<span><span class="co">#&gt;     url = {http://www.bioconductor.org/packages/smokingMouse},</span></span>
<span><span class="co">#&gt;     note = {https://github.com/LieberInstitute/smokingMouse/smokingMouse - R package version 0.99.5},</span></span>
<span><span class="co">#&gt;     doi = {10.18129/B9.bioc.smokingMouse},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="quick-start-to-using-smokingmouse">Quick start to using <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#quick-start-to-using-smokingmouse"></a>
</h2>
<p>To get started, please load the <em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/smokingMouse" class="external-link">"smokingMouse"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="smoking-mouse-datasets">smoking Mouse datasets<a class="anchor" aria-label="anchor" href="#smoking-mouse-datasets"></a>
</h2>
<p>The raw data was generated by LIBD researchers and is composed of
expression counts of genes, transcripts (txs), exons and exon-exon
junctions (jxns) across 208 mice samples (brain/blood; adult/pup;
nicotine-exposed/smoking-exposed/controls). The datasets available in
<em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
were generated by Daianna Gonzalez-Padilla. The human data was generated
by Semick, S.A. et al. (2018) in Mol Psychiatry, DOI: <a href="https://doi.org/10.1038/s41380-018-0223-1" class="external-link uri">https://doi.org/10.1038/s41380-018-0223-1</a> and it
contains the results of a DEA in adult and prenatal human brain samples
exposed to cigarette smoke.</p>
<div class="section level3">
<h3 id="description-of-the-datasets">Description of the datasets<a class="anchor" aria-label="anchor" href="#description-of-the-datasets"></a>
</h3>
<div class="section level4">
<h4 id="mouse-datasets">Mouse datasets:<a class="anchor" aria-label="anchor" href="#mouse-datasets"></a>
</h4>
<ul>
<li>They are 4 <em><a href="https://bioconductor.org/packages/3.17/RangedSummarizedExperiment" class="external-link">RangedSummarizedExperiment</a></em>
(RSE) objects that contain feature info in <code>rowData(RSE)</code> and
sample info in <code>colData(RSE)</code>.</li>
<li>Raw expression counts can be accessed with
<code>assays(RSE)$counts</code> and the lognorm counts (log2(cpm + 0.5)
for genes, exons and jxns; log2(tpm + 0.5) for txs) with
<code>assays(RSE)$logcounts</code>.</li>
</ul>
</div>
<div class="section level4">
<h4 id="human-datasets">Human datasets:<a class="anchor" aria-label="anchor" href="#human-datasets"></a>
</h4>
<ul>
<li>They are two data frames with the information of human genes and
some relevant statistical metrics of differential expression (DE).</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="data-specifics">Data specifics<a class="anchor" aria-label="anchor" href="#data-specifics"></a>
</h3>
<ul>
<li>
<em>‘rse_gene_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_gene object)
the gene RSE object contains raw and normalized expression data of 55401
mouse genes across 208 samples from brains and blood of healthy and
nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_tx_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_tx object) the tx
RSE object contains raw and normalized expression data of 142604 mouse
transcripts across 208 samples from brains and blood of healthy and
nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_exon_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_exon object)
the exon RSE object contains raw and normalized expression data of
447670 mouse exons across 208 samples from brains and blood of healthy
and nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_jx_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_jx object) the jx
RSE object contains raw and normalized expression data of 1436068 mouse
exon-exon junctions across 208 samples from brains and blood of healthy
and nicotine/smoking-exposed pup and adult mice.</li>
</ul>
<p>All the above datasets contain sample and feature information and
additional data of the results obtained in the filtering steps and the
DEA.</p>
<ul>
<li>
<em>‘de_genes_prenatal_human_brain_smoking.Rdata’</em>: (object with
the same name) data frame with DE (ctrls vs smoking-exposed samples)
data of 18067 genes in human prenatal brain samples exposed to cigarette
smoke.</li>
<li>
<em>‘de_genes_adult_human_brain_smoking.Rdata’</em>: (object with
the same name) data frame with DE (ctrls vs smoking-exposed samples)
data of 18067 genes in human adult brain samples exposed to cigarette
smoke.</li>
</ul>
</div>
<div class="section level3">
<h3 id="variables-of-mice-data">Variables of mice data<a class="anchor" aria-label="anchor" href="#variables-of-mice-data"></a>
</h3>
<p>Feature information in <code>rowData(RSE)</code> contains the
following variables:</p>
<ul>
<li>
<code>retained_after_feature_filtering</code>: Boolean variable that
equals TRUE if the feature passed the feature filtering based on
expression levels and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/blob/main/code/02_build_objects/02_build_objects.R" class="external-link">here</a>.</li>
<li>
<code>DE_in_adult_brain_nicotine</code>: Boolean variable that
equals TRUE if the feature is differentially expressed (DE) in adult
brain samples exposed to nicotine and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_adult_brain_smoking</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in adult brain
samples exposed to cigarette smoke and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_adult_blood_smoking</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in adult blood
samples exposed to cigarette smoke and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_pup_brain_nicotine</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in pup brain
samples exposed to nicotine and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_pup_brain_smoking</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in pup brain
samples exposed to cigarette smoke and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.
The rest of the variables are outputs of SPEAQeasy pipeline. See <a href="http://research.libd.org/SPEAQeasy/outputs.html" class="external-link">here</a> for a
description of them.</li>
</ul>
<p>Sample information in <code>colData(RSE)</code> contains the
following variables:</p>
<ul>
<li>The Quality Control (QC) variables
<code>sum</code>,<code>detected</code>,<code>subsets_Mito_sum</code>,
<code>subsets_Mito_detected</code>, <code>subsets_Mito_percent</code>,
<code>subsets_Ribo_sum</code>,<code>subsets_Ribo_detected</code> and
<code>subsets_Ribo_percent</code> are returned by
<code>addPerCellQC()</code>. See <a href="https://rdrr.io/bioc/scuttle/man/addPerCellQC.html" class="external-link">here</a> for
more info.</li>
<li>
<code>retained_after_QC_sample_filtering</code>: Boolean variable
that equals TRUE if the sample passed the sample filtering based on QC
metrics and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/blob/main/code/03_EDA/02_QC.R" class="external-link">here</a>.</li>
<li>
<code>retained_after_manual_sample_filtering</code>: Boolean
variable that equals TRUE if the sample passed the manual sample
filtering based on PCA plots and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/blob/main/code/03_EDA/03_PCA_MDS.R" class="external-link">here</a>
The rest of the variables are outputs of SPEAQeasy. See <a href="http://research.libd.org/SPEAQeasy/outputs.html" class="external-link">here</a> for
their description.</li>
</ul>
</div>
<div class="section level3">
<h3 id="variables-of-human-data">Variables of human data<a class="anchor" aria-label="anchor" href="#variables-of-human-data"></a>
</h3>
<p>Check <a href="https://github.com/LieberInstitute/Smoking_DLPFC_Devel" class="external-link">here</a>
to see the data generation and variables meaning.</p>
</div>
<div class="section level3">
<h3 id="downloading-the-data-with-smokingmouse">Downloading the data with <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#downloading-the-data-with-smokingmouse"></a>
</h3>
<p>Using <em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
you can download these R objects. They are hosted by <a href="http://bioconductor.org/" class="external-link">Bioconductor</a>’s <em><a href="https://bioconductor.org/packages/3.17/ExperimentHub" class="external-link">ExperimentHub</a></em>
(Morgan and Shepherd, 2023) resource. Below you can see how to obtain
these objects.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Connect to ExperimentHub</span></span>
<span><span class="va">ehub</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">###########      The following is just provisional code for the example.      ###########</span></span>
<span><span class="co">########### Once the package is approved and ready to access, the example will be correctly updated. ###########</span></span>
<span></span>
<span><span class="co">## Load the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/smokingMouse" class="external-link">"smokingMouse"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">########################</span></span>
<span><span class="co">#      Mouse data </span></span>
<span><span class="co">########################</span></span>
<span><span class="co"># ## Download the mouse gene data</span></span>
<span><span class="co"># rse_gene &lt;- function_to_access</span></span>
<span><span class="co"># ## This is a RangedSummarizedExperiment object</span></span>
<span><span class="co"># rse_gene</span></span>
<span></span>
<span><span class="co"># ## Note the memory size</span></span>
<span><span class="co"># lobstr::obj_size(rse_gene)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># ## Check sample info </span></span>
<span><span class="co"># head(colData(rse_gene), 3)</span></span>
<span><span class="co"># ## Check gene info</span></span>
<span><span class="co"># head(rowData(rse_gene), 3)</span></span>
<span><span class="co"># ## Access the original counts</span></span>
<span><span class="co"># original_counts &lt;- assays(rse_gene)$counts</span></span>
<span><span class="co"># ## Access the log normalized counts</span></span>
<span><span class="co"># logcounts &lt;- assays(rse_gene)$logcounts</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="co">########################</span></span>
<span><span class="co">#      Human data </span></span>
<span><span class="co">########################</span></span>
<span><span class="co"># ## Download the human gene data</span></span>
<span><span class="co"># de_genes_prenatal_human_brain_smoking &lt;- function_to_access</span></span>
<span><span class="co"># ## This is a data frame</span></span>
<span><span class="co"># de_genes_prenatal_human_brain_smoking</span></span>
<span></span>
<span><span class="co"># ## Note the memory size</span></span>
<span><span class="co"># lobstr::obj_size(de_genes_prenatal_human_brain_smoking)</span></span>
<span></span>
<span><span class="co">## Access data of human genes as normally do with data frames</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.17/smokingMouse" class="external-link">smokingMouse</a></em>
package and the <a href="https://github.com/LieberInstitute/smokingMouse_Indirects" class="external-link">smoking
mouse</a> project were made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li><em><a href="https://bioconductor.org/packages/3.17/AnnotationHub" class="external-link">AnnotationHub</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=cowplot" class="external-link">cowplot</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/edgeR" class="external-link">edgeR</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/ExperimentHub" class="external-link">ExperimentHub</a></em>
(Morgan and Shepherd, 2023)</li>
<li><em><a href="https://CRAN.R-project.org/package=jaffelab" class="external-link">jaffelab</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=ggrepel" class="external-link">ggrepel</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/scater" class="external-link">scater</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=rlang" class="external-link">rlang</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=gridExtra" class="external-link">gridExtra</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/variancePartition" class="external-link">variancePartition</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/limma" class="external-link">limma</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=VennDiagram" class="external-link">VennDiagram</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=biomartr" class="external-link">biomartr</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=Hmisc" class="external-link">Hmisc</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=R.utils" class="external-link">R.utils</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/clusterProfiler" class="external-link">clusterProfiler</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/org.Mm.eg.db" class="external-link">org.Mm.eg.db</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=pheatmap" class="external-link">pheatmap</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/GenomicRanges" class="external-link">GenomicRanges</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.17/rtracklayer" class="external-link">rtracklayer</a></em></li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.17/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2023-07-12 18:42:55 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 5.664 secs</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-07-12</span></span>
<span><span class="co">#&gt;  pandoc   3.1.1 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi            1.62.2     <span style="color: #555555;">2023-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationHub            3.8.0      <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  backports                1.4.1      <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bibtex                   0.5.1      <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Biobase                  2.60.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocFileCache            2.8.0      <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics             0.46.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager              1.30.21    <span style="color: #555555;">2023-06-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  BiocStyle              * 2.28.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocVersion              3.17.1     <span style="color: #555555;">2022-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings               2.68.1     <span style="color: #555555;">2023-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bit                      4.0.5      <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bit64                    4.0.5      <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bitops                   1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  blob                     1.2.4      <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bookdown                 0.34       <span style="color: #555555;">2023-05-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bslib                    0.5.0      <span style="color: #555555;">2023-06-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cachem                   1.0.8      <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cli                      3.6.1      <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  crayon                   1.5.2      <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  curl                     5.0.1      <span style="color: #555555;">2023-06-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  DBI                      1.1.3      <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  dbplyr                   2.3.3      <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  desc                     1.4.2      <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  digest                   0.6.33     <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  dplyr                    1.1.2      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ellipsis                 0.3.2      <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  evaluate                 0.21       <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ExperimentHub            2.8.0      <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  fansi                    1.0.4      <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fastmap                  1.1.1      <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  filelock                 1.0.2      <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fs                       1.6.2      <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  generics                 0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb             1.36.1     <span style="color: #555555;">2023-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData         1.2.10     <span style="color: #555555;">2023-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  glue                     1.6.2      <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  htmltools                0.5.5      <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  httpuv                   1.6.11     <span style="color: #555555;">2023-05-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  httr                     1.4.6      <span style="color: #555555;">2023-05-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  interactiveDisplayBase   1.38.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  IRanges                  2.34.1     <span style="color: #555555;">2023-06-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  jquerylib                0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  jsonlite                 1.8.7      <span style="color: #555555;">2023-06-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  KEGGREST                 1.40.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  knitr                    1.43       <span style="color: #555555;">2023-05-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  later                    1.3.1      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lifecycle                1.0.3      <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lubridate                1.9.2      <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  magrittr                 2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  memoise                  2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  mime                     0.12       <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pillar                   1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pkgdown                  <span style="color: #BB00BB; font-weight: bold;">2.0.7.9000</span> <span style="color: #555555;">2023-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (r-lib/pkgdown@39083f3)</span></span></span>
<span><span class="co">#&gt;  plyr                     1.8.8      <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  png                      0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  promises                 1.2.0.1    <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  purrr                    1.0.1      <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  R6                       2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ragg                     1.2.5      <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rappdirs                 0.3.3      <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Rcpp                     1.0.11     <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RCurl                    1.98-1.12  <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RefManageR             * 1.4.0      <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rlang                    1.1.1      <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                2.23       <span style="color: #555555;">2023-07-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rprojroot                2.0.3      <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RSQLite                  2.3.1      <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  S4Vectors                0.38.1     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                     0.4.6      <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  sessioninfo            * 1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  shiny                    1.7.4.1    <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  smokingMouse           * 0.99.5     <span style="color: #555555;">2023-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  stringi                  1.7.12     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  stringr                  1.5.0      <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts              1.0.4      <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  textshaping              0.3.6      <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tibble                   3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidyselect               1.2.0      <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  timechange               0.2.0      <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  utf8                     1.2.3      <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  vctrs                    0.6.3      <span style="color: #555555;">2023-06-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  withr                    2.5.0      <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xfun                     0.39       <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xml2                     1.3.5      <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xtable                   1.8-4      <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  XVector                  0.40.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                     2.3.7      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                 1.46.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.17/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.23. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[2]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2023experimenthub"></a><a href="#cite-morgan2023experimenthub">[3]</a><cite>
M. Morgan and L. Shepherd. <em>ExperimentHub: Client to access
ExperimentHub resources</em>. R package version 2.8.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.ExperimentHub" class="external-link">10.18129/B9.bioc.ExperimentHub</a>.
URL:
<a href="https://bioconductor.org/packages/ExperimentHub" class="external-link">https://bioconductor.org/packages/ExperimentHub</a>.</cite>
</p>
<p>
<a id="bib-ole2023biocstyle"></a><a href="#cite-ole2023biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.28.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2023language"></a><a href="#cite-2023language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2023. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2. 2021. URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2023knitr"></a><a href="#cite-xie2023knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.43. 2023. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Daianna Gonzalez-Padilla.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
