<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to smokingMouse • smokingMouse</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Introduction to smokingMouse">
<meta property="og:description" content="smokingMouse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">smokingMouse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/smokingMouse.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/smokingMouse/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to smokingMouse</h1>
                        <h4 data-toc-skip class="author">Daianna
Gonzalez-Padilla</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development
(LIBD)<br><a class="author_email" href="mailto:#"></a><a href="mailto:glezdaianna@gmail.com" class="email">glezdaianna@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">21 July 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/smokingMouse/blob/HEAD/vignettes/smokingMouse.Rmd" class="external-link"><code>vignettes/smokingMouse.Rmd</code></a></small>
      <div class="hidden name"><code>smokingMouse.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Welcome to the <code>smokingMouse</code> project. In this vignette
we’ll show you how to access the smokingMouse LIBD datasets . You can
find the analysis code and the data generation in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects" class="external-link">here</a>.</p>
<div class="section level3">
<h3 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h3>
<p>The main motivation to create this bioconductor package was to
provide public and free access to all RNA-seq datasets that were
generated for the smokingMouse project, containing many variables of
interest that make it possible to answer a wide range of biological
questions related to smoking and nicotine effects in mice.</p>
</div>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>This bulk RNA-sequencing project consisted of a differential
expression analysis (DEA) involving 4 data types: genes, exons,
transcripts and junctions. The main goal of this study was to explore
the effects of prenatal exposure to maternal smoking and nicotine
exposures on the developing mouse brain. As secondary objectives, this
work evaluated: 1) the affected genes by each exposure on the adult
female brain in order to compare offspring and adult results and 2) the
effects of smoking on adult blood and brain to search for overlapping
biomarkers in both tissues. Finally, DEGs identified in mice were
compared against previously published results from a human study
(Semick, S.A. et al. (2018)).</p>
<p>The next table summarizes the analyses done at each level.</p>
<figure><img src="http://research.libd.org/smokingMouse/reference/figures/Table_of_Analyses.png" align="center" width="800px"><figcaption style="color: gray; line-height: 0.94; text-align: justify"><p><font size="-1.5"><b>Table 1</b>: <b>1. Data preparation</b>: in this
first step, counts of genes, exons and junctions were normalized to CPM
and scaled; transcript expression values were only scaled since they
were already in TMP. Then, low-expression features were removed using
the indicated methods and samples were separated by tissue and age in
order to create subsets of the data for downstream analyses. <b>2.
Exploratory Data Analysis</b>: QC metrics of the samples were examined
and used to filter them; sample level effects were explored through
dimensionality reduction methods and rare samples in PCA plots were
manually removed from the datasets; gene level effects were evaluated
with analyses of explanatory variables and variance partition. <b>3.
Differential Expression Analysis</b>: with the relevant variables
identified in the previous steps, the DEA was performed at the gene
level for nicotine and smoking, adult and pup, and blood and brain
samples, and for 3 models: the naive one modeled <span style="font-family: monospace">~Group + batch effects</span>, the
adjusted model modeled <span style="font-family: monospace">~Group +
Pregnancy + batch effects</span> for adults and <span style="font-family: monospace">~Group + Sex + batch effects</span> for
pups, and the interaction model <span style="font-family: monospace">~Group*Pregnancy + batch effects</span>
for adults and <span style="font-family: monospace">~Group*Sex + batch
effects</span> for pups; DEA on the rest of the levels was performed for
pups only and using the adjusted model. After that, signals of the
features in nicotine and smoking were compared, as well as the signals
of exons and txs vs the effects of their genes, and genes’ signals were
additionally compared in the different tissues, ages, models and species
(vs human data of a previous study). All resultant DEG and DE features
(and their genes) were quantified and compared based on their experiment
(nic/smo) and direction of regulation (up/down); DEG were further
compared against genes of DE exons and txs; mouse genes were also
compared with human genes affected by cigarette smoke or associated with
TUD. <b>4. Gene Ontology and KEGG</b>: taking the DEG and the genes of
DE txs and exons, GO &amp; KEGG analyses were done and the expression
levels of genes that participate in brain development related processes
were explored. <b>5. DE feature visualization</b>: DEG counts were
represented in heatmaps in order to distinguish the groups of up and
down-regulated genes. <b>6. Junction annotation</b>: for novel DE jxns
of unknown gene, their nearest, preceding and following genes were
determined.</font></p>
<p><font size="0.8">Abbreviations: Jxn: junction; Tx: transcript; CPM:
counts per million; TPM: transcripts per million; TMM: Trimmed Mean of
M-Values; TMMwsp: TMM with singleton pairing; EDA: exploratory data
analysis; QC: quality control; ribo: ribosomal; mt: mitochondrial; PCA:
Principal Component Analysis; PC: principal component; MDS:
Multidimensional Scaling; DEA: differential expression analysis; DE:
differential expression/differentially expressed; DEG: differentially
expressed genes; GO: Gene Ontology; KEGG: Kyoto Encyclopedia of Genes
and Genomes; TUD: tobacco use disorder.</font></p>
</figcaption></figure>
</div>
<div class="section level3">
<h3 id="study-design">Study design<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h3>
<figure><img src="http://research.libd.org/smokingMouse/reference/figures/Study_design.png" align="center" width="800px"><figcaption style="color: gray; line-height: 0.88; text-align: justify"><p><font size="-1.5"><b>Figure 1</b>: <b>Experimental design of the
study</b>. <b>A)</b> 36 pregnant dams and 35 non-pregnant female adult
mice were either administered nicotine by intraperitoneal injection (IP;
n=12), exposed to cigarette smoke in smoking chambers (n=24), or
controls (n=35; 11 nicotine controls and 24 smoking controls). A total
of 137 pups were born to pregnant dams: 19 were born to mice that were
administered nicotine, 46 to mice exposed to cigarette smoke and the
remaining 72 to control mice (23 to nicotine controls and 49 to smoking
controls). Samples from frontal cortices of P0 pups and adults were
obtained, as well as blood samples from smoking-exposed and smoking
control adults. <b>B)</b> RNA was extracted, RNA-seq libraries were
prepared and sequenced to obtain expression counts for genes, exons,
transcripts and exon-exon junctions.</font></p>
</figcaption></figure>
</div>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-smokingmouse">Install <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#install-smokingmouse"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
is a <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages.
<code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"smokingMouse"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
is based on many other packages and in particular in those that have
implemented the infrastructure needed for dealing with RNA-seq data and
differential expression. That is, packages like <em><a href="https://bioconductor.org/packages/3.18/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
and <em><a href="https://bioconductor.org/packages/3.18/limma" class="external-link">limma</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help: remember to use the
<code>smokingMouse</code> tag and check <a href="https://support.bioconductor.org/tag/smokingMouse/" class="external-link">the older
posts</a>. Other alternatives are available such as creating GitHub
issues and tweeting. However, please note that if you want to receive
help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="citing-smokingmouse">Citing <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#citing-smokingmouse"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"smokingMouse"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'smokingMouse' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Gonzalez-Padilla D, Collado-Torres L (2023). _Provides access to</span></span>
<span><span class="co">#&gt;   smokingMouse project data _. doi:10.18129/B9.bioc.smokingMouse</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.18129/B9.bioc.smokingMouse&gt;,</span></span>
<span><span class="co">#&gt;   https://github.com/LieberInstitute/smokingMouse/smokingMouse - R</span></span>
<span><span class="co">#&gt;   package version 0.99.9,</span></span>
<span><span class="co">#&gt;   &lt;http://www.bioconductor.org/packages/smokingMouse&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {Provides access to smokingMouse project data },</span></span>
<span><span class="co">#&gt;     author = {Daianna Gonzalez-Padilla and Leonardo Collado-Torres},</span></span>
<span><span class="co">#&gt;     year = {2023},</span></span>
<span><span class="co">#&gt;     url = {http://www.bioconductor.org/packages/smokingMouse},</span></span>
<span><span class="co">#&gt;     note = {https://github.com/LieberInstitute/smokingMouse/smokingMouse - R package version 0.99.9},</span></span>
<span><span class="co">#&gt;     doi = {10.18129/B9.bioc.smokingMouse},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="quick-start-to-using-smokingmouse">Quick start to using <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#quick-start-to-using-smokingmouse"></a>
</h2>
<p>To get started, please load the <em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/smokingMouse" class="external-link">"smokingMouse"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="smoking-mouse-datasets">smoking Mouse datasets<a class="anchor" aria-label="anchor" href="#smoking-mouse-datasets"></a>
</h2>
<p>The raw data was generated by LIBD researchers and is composed of
expression counts of genes, transcripts (txs), exons and exon-exon
junctions (jxns) across 208 mice samples (brain/blood; adult/pup;
nicotine-exposed/smoking-exposed/controls). The datasets available in
<em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
were generated by Daianna Gonzalez-Padilla. The human data was generated
by Semick, S.A. et al. (2018) in Mol Psychiatry, DOI: <a href="https://doi.org/10.1038/s41380-018-0223-1" class="external-link uri">https://doi.org/10.1038/s41380-018-0223-1</a> and it
contains the results of a DEA in adult and prenatal human brain samples
exposed to cigarette smoke.</p>
<div class="section level3">
<h3 id="description-of-the-datasets">Description of the datasets<a class="anchor" aria-label="anchor" href="#description-of-the-datasets"></a>
</h3>
<div class="section level4">
<h4 id="mouse-datasets">Mouse datasets:<a class="anchor" aria-label="anchor" href="#mouse-datasets"></a>
</h4>
<ul>
<li>They are 4 <em><a href="https://bioconductor.org/packages/3.18/RangedSummarizedExperiment" class="external-link">RangedSummarizedExperiment</a></em>
(RSE) objects that contain feature info in <code>rowData(RSE)</code> and
sample info in <code>colData(RSE)</code>.</li>
<li>Raw expression counts can be accessed with
<code>assays(RSE)$counts</code> and the lognorm counts (log2(cpm + 0.5)
for genes, exons and jxns; log2(tpm + 0.5) for txs) with
<code>assays(RSE)$logcounts</code>.</li>
</ul>
</div>
<div class="section level4">
<h4 id="human-datasets">Human datasets:<a class="anchor" aria-label="anchor" href="#human-datasets"></a>
</h4>
<ul>
<li>They are two data frames with the information of human genes and
some relevant statistical metrics of differential expression (DE).</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="data-specifics">Data specifics<a class="anchor" aria-label="anchor" href="#data-specifics"></a>
</h3>
<ul>
<li>
<em>‘rse_gene_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_gene object)
the gene RSE object contains raw and normalized expression data of 55401
mouse genes across 208 samples from brains and blood of healthy and
nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_tx_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_tx object) the tx
RSE object contains raw and normalized expression data of 142604 mouse
transcripts across 208 samples from brains and blood of healthy and
nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_exon_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_exon object)
the exon RSE object contains raw and normalized expression data of
447670 mouse exons across 208 samples from brains and blood of healthy
and nicotine/smoking-exposed pup and adult mice.</li>
<li>
<em>‘rse_jx_mouse_RNAseq_nic-smo.Rdata’</em>: (rse_jx object) the jx
RSE object contains raw and normalized expression data of 1436068 mouse
exon-exon junctions across 208 samples from brains and blood of healthy
and nicotine/smoking-exposed pup and adult mice.</li>
</ul>
<p>All the above datasets contain sample and feature information and
additional data of the results obtained in the filtering steps and the
DEA.</p>
<ul>
<li>
<em>‘de_genes_prenatal_human_brain_smoking.Rdata’</em>: (object with
the same name) data frame with DE (ctrls vs smoking-exposed samples)
data of 18067 genes in human prenatal brain samples exposed to cigarette
smoke.</li>
<li>
<em>‘de_genes_adult_human_brain_smoking.Rdata’</em>: (object with
the same name) data frame with DE (ctrls vs smoking-exposed samples)
data of 18067 genes in human adult brain samples exposed to cigarette
smoke.</li>
</ul>
</div>
<div class="section level3">
<h3 id="variables-of-mice-data">Variables of mice data<a class="anchor" aria-label="anchor" href="#variables-of-mice-data"></a>
</h3>
<p>Feature information in <code>rowData(RSE)</code> contains the
following variables:</p>
<ul>
<li>
<code>retained_after_feature_filtering</code>: Boolean variable that
equals TRUE if the feature passed the feature filtering based on
expression levels and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/blob/main/code/02_build_objects/02_build_objects.R" class="external-link">here</a>.</li>
<li>
<code>DE_in_adult_brain_nicotine</code>: Boolean variable that
equals TRUE if the feature is differentially expressed (DE) in adult
brain samples exposed to nicotine and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_adult_brain_smoking</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in adult brain
samples exposed to cigarette smoke and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_adult_blood_smoking</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in adult blood
samples exposed to cigarette smoke and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_pup_brain_nicotine</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in pup brain
samples exposed to nicotine and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.</li>
<li>
<code>DE_in_pup_brain_smoking</code>: Boolean variable that equals
TRUE if the feature is differentially expressed (DE) in pup brain
samples exposed to cigarette smoke and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/tree/main/code/04_DEA" class="external-link">here</a>.
The rest of the variables are outputs of SPEAQeasy pipeline. See <a href="http://research.libd.org/SPEAQeasy/outputs.html" class="external-link">here</a> for a
description of them.</li>
</ul>
<p>Sample information in <code>colData(RSE)</code> contains the
following variables:</p>
<ul>
<li>The Quality Control (QC) variables
<code>sum</code>,<code>detected</code>,<code>subsets_Mito_sum</code>,
<code>subsets_Mito_detected</code>, <code>subsets_Mito_percent</code>,
<code>subsets_Ribo_sum</code>,<code>subsets_Ribo_detected</code> and
<code>subsets_Ribo_percent</code> are returned by
<code>addPerCellQC()</code>. See <a href="https://rdrr.io/bioc/scuttle/man/addPerCellQC.html" class="external-link">here</a> for
more info.</li>
<li>
<code>retained_after_QC_sample_filtering</code>: Boolean variable
that equals TRUE if the sample passed the sample filtering based on QC
metrics and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/blob/main/code/03_EDA/02_QC.R" class="external-link">here</a>.</li>
<li>
<code>retained_after_manual_sample_filtering</code>: Boolean
variable that equals TRUE if the sample passed the manual sample
filtering based on PCA plots and FALSE if not. Check code in <a href="https://github.com/LieberInstitute/smokingMouse_Indirects/blob/main/code/03_EDA/03_PCA_MDS.R" class="external-link">here</a>
The rest of the variables are outputs of SPEAQeasy. See <a href="http://research.libd.org/SPEAQeasy/outputs.html" class="external-link">here</a> for
their description.</li>
</ul>
</div>
<div class="section level3">
<h3 id="variables-of-human-data">Variables of human data<a class="anchor" aria-label="anchor" href="#variables-of-human-data"></a>
</h3>
<p>Check <a href="https://github.com/LieberInstitute/Smoking_DLPFC_Devel" class="external-link">here</a>
to see the data generation and variables meaning.</p>
</div>
<div class="section level3">
<h3 id="downloading-the-data-with-smokingmouse">Downloading the data with <code>smokingMouse</code><a class="anchor" aria-label="anchor" href="#downloading-the-data-with-smokingmouse"></a>
</h3>
<p>Using <em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
you can download these R objects. They are hosted by <a href="http://bioconductor.org/" class="external-link">Bioconductor</a>’s <em><a href="https://bioconductor.org/packages/3.18/ExperimentHub" class="external-link">ExperimentHub</a></em>
(Morgan and Shepherd, 2023) resource. Below you can see how to obtain
these objects.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load ExperimentHub for downloading the data</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"ExperimentHub"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationHub</span></span>
<span><span class="co">#&gt; Loading required package: BiocFileCache</span></span>
<span><span class="co">#&gt; Loading required package: dbplyr</span></span>
<span></span>
<span><span class="co">## Connect to ExperimentHub</span></span>
<span><span class="va">ehub</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load the datasets of the package</span></span>
<span><span class="va">myfiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ehub</span>, <span class="st">"smokingMouse"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Resulting smokingMouse files from our ExperimentHub query</span></span>
<span><span class="va">myfiles</span></span>
<span><span class="co">#&gt; ExperimentHub with 6 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2023-07-21</span></span>
<span><span class="co">#&gt; # $dataprovider: Lieber Institute for Brain Development (LIBD)</span></span>
<span><span class="co">#&gt; # $species: Mus musculus, Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: RangedSummarizedExperiment, GenomicRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["EH8313"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;            title                                </span></span>
<span><span class="co">#&gt;   EH8313 | rse_gene_mouse_RNAseq_nic-smo        </span></span>
<span><span class="co">#&gt;   EH8314 | rse_tx_mouse_RNAseq_nic-smo          </span></span>
<span><span class="co">#&gt;   EH8315 | rse_jx_mouse_RNAseq_nic-smo          </span></span>
<span><span class="co">#&gt;   EH8316 | rse_exon_mouse_RNAseq_nic-smo        </span></span>
<span><span class="co">#&gt;   EH8317 | de_genes_prenatal_human_brain_smoking</span></span>
<span><span class="co">#&gt;   EH8318 | de_genes_adult_human_brain_smoking</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load SummarizedExperiment which defines the class container for the data</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">"SummarizedExperiment"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:ExperimentHub':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     cache</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:AnnotationHub':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     cache</span></span>
<span></span>
<span><span class="co">######################</span></span>
<span><span class="co">#     Mouse data </span></span>
<span><span class="co">######################</span></span>
<span><span class="va">myfiles</span><span class="op">[</span><span class="st">"EH8313"</span><span class="op">]</span></span>
<span><span class="co">#&gt; ExperimentHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2023-07-21</span></span>
<span><span class="co">#&gt; # names(): EH8313</span></span>
<span><span class="co">#&gt; # package(): smokingMouse</span></span>
<span><span class="co">#&gt; # $dataprovider: Lieber Institute for Brain Development (LIBD)</span></span>
<span><span class="co">#&gt; # $species: Mus musculus</span></span>
<span><span class="co">#&gt; # $rdataclass: RangedSummarizedExperiment</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2023-07-21</span></span>
<span><span class="co">#&gt; # $title: rse_gene_mouse_RNAseq_nic-smo</span></span>
<span><span class="co">#&gt; # $description: RangedSummarizedExperiment of bulk RNA-seq data from mouse b...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 10090</span></span>
<span><span class="co">#&gt; # $genome: GRCm38</span></span>
<span><span class="co">#&gt; # $sourcetype: GTF</span></span>
<span><span class="co">#&gt; # $sourceurl: https://bioconductor.org/packages/smokingMouse</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("ExpressionData", "LIBD", "mouse", "Mus_musculus_Data",</span></span>
<span><span class="co">#&gt; #   "nicotine", "RNASeqData", "smoking") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["EH8313"]]'</span></span>
<span></span>
<span><span class="co">## Download the mouse gene data</span></span>
<span><span class="co">#  EH8313 | rse_gene_mouse_RNAseq_nic-smo</span></span>
<span><span class="va">rse_gene</span> <span class="op">&lt;-</span> <span class="va">myfiles</span><span class="op">[[</span><span class="st">"EH8313"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; see ?smokingMouse and browseVignettes('smokingMouse') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span></span>
<span><span class="co">## This is a RangedSummarizedExperiment object</span></span>
<span><span class="va">rse_gene</span></span>
<span><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 55401 208 </span></span>
<span><span class="co">#&gt; metadata(1): Obtained_from</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(55401): ENSMUSG00000102693.1 ENSMUSG00000064842.1 ...</span></span>
<span><span class="co">#&gt;   ENSMUSG00000064371.1 ENSMUSG00000064372.1</span></span>
<span><span class="co">#&gt; rowData names(13): Length gencodeID ... DE_in_pup_brain_nicotine</span></span>
<span><span class="co">#&gt;   DE_in_pup_brain_smoking</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; colData names(71): SAMPLE_ID FQCbasicStats ...</span></span>
<span><span class="co">#&gt;   retained_after_QC_sample_filtering</span></span>
<span><span class="co">#&gt;   retained_after_manual_sample_filtering</span></span>
<span></span>
<span><span class="co">## Optionally check the memory size</span></span>
<span><span class="co"># lobstr::obj_size(rse_gene)</span></span>
<span><span class="co"># 159.68 MB</span></span>
<span></span>
<span><span class="co">## Check sample info </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 3 rows and 71 columns</span></span>
<span><span class="co">#&gt;     SAMPLE_ID FQCbasicStats perBaseQual perTileQual  perSeqQual perBaseContent</span></span>
<span><span class="co">#&gt;   &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt; 1 Sample_2914          PASS        PASS        PASS        PASS      FAIL/WARN</span></span>
<span><span class="co">#&gt; 2 Sample_4042          PASS        PASS        PASS        PASS      FAIL/WARN</span></span>
<span><span class="co">#&gt; 3 Sample_4043          PASS        PASS        PASS        PASS      FAIL/WARN</span></span>
<span><span class="co">#&gt;     GCcontent    Ncontent SeqLengthDist SeqDuplication OverrepSeqs</span></span>
<span><span class="co">#&gt;   &lt;character&gt; &lt;character&gt;   &lt;character&gt;    &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; 1        WARN        PASS          WARN           FAIL        PASS</span></span>
<span><span class="co">#&gt; 2        WARN        PASS          WARN           FAIL        PASS</span></span>
<span><span class="co">#&gt; 3        WARN        PASS          WARN           FAIL        PASS</span></span>
<span><span class="co">#&gt;   AdapterContent KmerContent SeqLength_R1 percentGC_R1 phred15-19_R1</span></span>
<span><span class="co">#&gt;      &lt;character&gt; &lt;character&gt;  &lt;character&gt;  &lt;character&gt;   &lt;character&gt;</span></span>
<span><span class="co">#&gt; 1           PASS          NA       75-151           49          37.0</span></span>
<span><span class="co">#&gt; 2           PASS          NA       75-151           48          37.0</span></span>
<span><span class="co">#&gt; 3           PASS          NA       75-151           49          37.0</span></span>
<span><span class="co">#&gt;   phred65-69_R1 phred115-119_R1 phred150-151_R1 phredGT30_R1 phredGT35_R1</span></span>
<span><span class="co">#&gt;     &lt;character&gt;     &lt;character&gt;     &lt;character&gt;    &lt;numeric&gt;    &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1          37.0            37.0            37.0           NA           NA</span></span>
<span><span class="co">#&gt; 2          37.0            37.0            37.0           NA           NA</span></span>
<span><span class="co">#&gt; 3          37.0            37.0            37.0           NA           NA</span></span>
<span><span class="co">#&gt;   Adapter65-69_R1 Adapter100-104_R1 Adapter140_R1 SeqLength_R2 percentGC_R2</span></span>
<span><span class="co">#&gt;         &lt;numeric&gt;         &lt;numeric&gt;     &lt;numeric&gt;  &lt;character&gt;  &lt;character&gt;</span></span>
<span><span class="co">#&gt; 1     0.000108294       0.000260890    0.00174971       75-151           49</span></span>
<span><span class="co">#&gt; 2     0.000210067       0.000352780    0.00154716       75-151           48</span></span>
<span><span class="co">#&gt; 3     0.000134434       0.000284284    0.00191475       75-151           49</span></span>
<span><span class="co">#&gt;   phred15-19_R2 phred65-69_R2 phred115-119_R2 phred150-151_R2 phredGT30_R2</span></span>
<span><span class="co">#&gt;     &lt;character&gt;   &lt;character&gt;     &lt;character&gt;     &lt;character&gt;    &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1          37.0          37.0            37.0            37.0           NA</span></span>
<span><span class="co">#&gt; 2          37.0          37.0            37.0            37.0           NA</span></span>
<span><span class="co">#&gt; 3          37.0          37.0            37.0            37.0           NA</span></span>
<span><span class="co">#&gt;   phredGT35_R2 Adapter65-69_R2 Adapter100-104_R2 Adapter140_R2 ERCCsumLogErr</span></span>
<span><span class="co">#&gt;      &lt;numeric&gt;       &lt;numeric&gt;         &lt;numeric&gt;     &lt;numeric&gt;     &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1           NA     0.000276104       0.000486875    0.00132235      -58.2056</span></span>
<span><span class="co">#&gt; 2           NA     0.000326771       0.000574851    0.00137044      -81.6359</span></span>
<span><span class="co">#&gt; 3           NA     0.000330534       0.000521084    0.00153550      -99.5348</span></span>
<span><span class="co">#&gt;                  bamFile   trimmed  numReads numMapped numUnmapped</span></span>
<span><span class="co">#&gt;              &lt;character&gt; &lt;logical&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1 Sample_2914_sorted.bam     FALSE  89386472  87621022     1765450</span></span>
<span><span class="co">#&gt; 2 Sample_4042_sorted.bam     FALSE  59980794  58967812     1012982</span></span>
<span><span class="co">#&gt; 3 Sample_4043_sorted.bam     FALSE  64864732  63961359      903373</span></span>
<span><span class="co">#&gt;   overallMapRate concordMapRate totalMapped mitoMapped  mitoRate</span></span>
<span><span class="co">#&gt;        &lt;numeric&gt;      &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1         0.9802         0.9748    87143087   10039739  0.103308</span></span>
<span><span class="co">#&gt; 2         0.9831         0.9709    58215746    7453208  0.113497</span></span>
<span><span class="co">#&gt; 3         0.9861         0.9811    62983384    8307331  0.116528</span></span>
<span><span class="co">#&gt;   totalAssignedGene  rRNA_rate      Tissue         Age         Sex        Expt</span></span>
<span><span class="co">#&gt;           &lt;numeric&gt;  &lt;numeric&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; 1          0.761378 0.00396315       Brain       Adult           F    Nicotine</span></span>
<span><span class="co">#&gt; 2          0.754444 0.00301119       Brain       Adult           F    Nicotine</span></span>
<span><span class="co">#&gt; 3          0.757560 0.00288706       Brain       Adult           F    Nicotine</span></span>
<span><span class="co">#&gt;          Group    Pregnant       plate    location concentration      medium</span></span>
<span><span class="co">#&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;   &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; 1 Experimental           0      Plate2         C12         165.9       Water</span></span>
<span><span class="co">#&gt; 2      Control           0      Plate1          B4         122.6       Water</span></span>
<span><span class="co">#&gt; 3      Control           0      Plate2          C9         128.5       Water</span></span>
<span><span class="co">#&gt;          date   Pregnancy    flowcell       sum  detected subsets_Mito_sum</span></span>
<span><span class="co">#&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;        &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1          NA          No   HKCMHDSXX  37119948     24435          2649559</span></span>
<span><span class="co">#&gt; 2          NA          No   HKCG7DSXX  24904754     23656          1913803</span></span>
<span><span class="co">#&gt; 3          NA          No   HKCMHDSXX  27083602     23903          2180712</span></span>
<span><span class="co">#&gt;   subsets_Mito_detected subsets_Mito_percent subsets_Ribo_sum</span></span>
<span><span class="co">#&gt;               &lt;numeric&gt;            &lt;numeric&gt;        &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1                    26              7.13783           486678</span></span>
<span><span class="co">#&gt; 2                    31              7.68449           319445</span></span>
<span><span class="co">#&gt; 3                    31              8.05178           338277</span></span>
<span><span class="co">#&gt;   subsets_Ribo_detected subsets_Ribo_percent retained_after_QC_sample_filtering</span></span>
<span><span class="co">#&gt;               &lt;numeric&gt;            &lt;numeric&gt;                          &lt;logical&gt;</span></span>
<span><span class="co">#&gt; 1                    11              1.31110                               TRUE</span></span>
<span><span class="co">#&gt; 2                    13              1.28267                               TRUE</span></span>
<span><span class="co">#&gt; 3                    14              1.24901                               TRUE</span></span>
<span><span class="co">#&gt;   retained_after_manual_sample_filtering</span></span>
<span><span class="co">#&gt;                                &lt;logical&gt;</span></span>
<span><span class="co">#&gt; 1                                   TRUE</span></span>
<span><span class="co">#&gt; 2                                   TRUE</span></span>
<span><span class="co">#&gt; 3                                   TRUE</span></span>
<span></span>
<span><span class="co">## Check gene info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 3 rows and 13 columns</span></span>
<span><span class="co">#&gt;                         Length            gencodeID          ensemblID</span></span>
<span><span class="co">#&gt;                      &lt;integer&gt;          &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693.1      1070 ENSMUSG00000102693.1 ENSMUSG00000102693</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1       110 ENSMUSG00000064842.1 ENSMUSG00000064842</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5      6094 ENSMUSG00000051951.5 ENSMUSG00000051951</span></span>
<span><span class="co">#&gt;                           gene_type    EntrezID          Symbol meanExprs</span></span>
<span><span class="co">#&gt;                         &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693.1            TEC       71042 MGI:MGI:1918292   0.00000</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1          snRNA          NA              NA   0.00000</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5 protein_coding      497097 MGI:MGI:3528744   7.94438</span></span>
<span><span class="co">#&gt;                      retained_after_feature_filtering DE_in_adult_blood_smoking</span></span>
<span><span class="co">#&gt;                                             &lt;logical&gt;                 &lt;logical&gt;</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693.1                            FALSE                     FALSE</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1                            FALSE                     FALSE</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5                             TRUE                     FALSE</span></span>
<span><span class="co">#&gt;                      DE_in_adult_brain_nicotine DE_in_adult_brain_smoking</span></span>
<span><span class="co">#&gt;                                       &lt;logical&gt;                 &lt;logical&gt;</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693.1                      FALSE                     FALSE</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1                      FALSE                     FALSE</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5                      FALSE                     FALSE</span></span>
<span><span class="co">#&gt;                      DE_in_pup_brain_nicotine DE_in_pup_brain_smoking</span></span>
<span><span class="co">#&gt;                                     &lt;logical&gt;               &lt;logical&gt;</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693.1                    FALSE                   FALSE</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1                    FALSE                   FALSE</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5                    FALSE                   FALSE</span></span>
<span></span>
<span><span class="co">## Access the original counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "matrix" "array"</span></span>
<span></span>
<span><span class="co">## Access the log normalized counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "matrix" "array"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 55401   208</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                           [,1]      [,2]      [,3]</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693.1 -5.985331 -5.985331 -5.985331</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842.1 -5.985331 -5.985331 -5.985331</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951.5  4.509114  4.865612  4.944597</span></span>
<span></span>
<span></span>
<span><span class="co">######################</span></span>
<span><span class="co">#     Human data </span></span>
<span><span class="co">######################</span></span>
<span><span class="va">myfiles</span><span class="op">[</span><span class="st">"EH8318"</span><span class="op">]</span></span>
<span><span class="co">#&gt; ExperimentHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2023-07-21</span></span>
<span><span class="co">#&gt; # names(): EH8318</span></span>
<span><span class="co">#&gt; # package(): smokingMouse</span></span>
<span><span class="co">#&gt; # $dataprovider: Lieber Institute for Brain Development (LIBD)</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GenomicRanges</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2023-07-21</span></span>
<span><span class="co">#&gt; # $title: de_genes_adult_human_brain_smoking</span></span>
<span><span class="co">#&gt; # $description: GRanges with the information of the differentialy expressed ...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh37</span></span>
<span><span class="co">#&gt; # $sourcetype: GTF</span></span>
<span><span class="co">#&gt; # $sourceurl: https://bioconductor.org/packages/smokingMouse</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("ExpressionData", "LIBD", "mouse", "Mus_musculus_Data",</span></span>
<span><span class="co">#&gt; #   "nicotine", "RNASeqData", "smoking") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["EH8318"]]'</span></span>
<span></span>
<span><span class="co">## Download the human gene data</span></span>
<span><span class="co"># EH8318 | de_genes_adult_human_brain_smoking</span></span>
<span><span class="va">de_genes_prenatal_human_brain_smoking</span> <span class="op">&lt;-</span> <span class="va">myfiles</span><span class="op">[[</span><span class="st">"EH8318"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; see ?smokingMouse and browseVignettes('smokingMouse') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span></span>
<span><span class="co">## This is a GRanges object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">de_genes_prenatal_human_brain_smoking</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GRanges"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "GenomicRanges"</span></span>
<span><span class="va">de_genes_prenatal_human_brain_smoking</span></span>
<span><span class="co">#&gt; GRanges object with 18067 ranges and 9 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames              ranges strand |    Length      Symbol</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000019169     chr2 119699742-119752236      + |      2079       MARCO</span></span>
<span><span class="co">#&gt;   ENSG00000260400    chr10   70458257-70460551      + |      2295            </span></span>
<span><span class="co">#&gt;   ENSG00000011201     chrX     8496915-8700227      - |      7131        KAL1</span></span>
<span><span class="co">#&gt;   ENSG00000068438     chrX   48334541-48344752      + |      2740       FTSJ1</span></span>
<span><span class="co">#&gt;   ENSG00000151229    chr12   40148823-40499891      - |     10027     SLC2A13</span></span>
<span><span class="co">#&gt;               ...      ...                 ...    ... .       ...         ...</span></span>
<span><span class="co">#&gt;   ENSG00000141556    chr17   80709940-80900724      + |     10472        TBCD</span></span>
<span><span class="co">#&gt;   ENSG00000125804    chr20   26035291-26073683      + |      6332     FAM182A</span></span>
<span><span class="co">#&gt;   ENSG00000228998    chr15   90818266-90820841      + |      2576            </span></span>
<span><span class="co">#&gt;   ENSG00000149636    chr20   35380194-35402221      - |      2739        DSN1</span></span>
<span><span class="co">#&gt;   ENSG00000122644     chr7   12726481-12730559      + |      3561       ARL4A</span></span>
<span><span class="co">#&gt;                    EntrezID        logFC   AveExpr            t     P.Value</span></span>
<span><span class="co">#&gt;                   &lt;integer&gt;    &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000019169      8685   -1.6032766  -1.80183     -6.14514 4.66989e-09</span></span>
<span><span class="co">#&gt;   ENSG00000260400      &lt;NA&gt;    0.1515813   1.17142      4.09836 6.18298e-05</span></span>
<span><span class="co">#&gt;   ENSG00000011201      3730    0.1423143   4.24576      4.09392 6.29277e-05</span></span>
<span><span class="co">#&gt;   ENSG00000068438     24140   -0.0495086   4.30660     -4.05975 7.20166e-05</span></span>
<span><span class="co">#&gt;   ENSG00000151229    114134    0.0842742   7.02625      4.00115 9.05839e-05</span></span>
<span><span class="co">#&gt;               ...       ...          ...       ...          ...         ...</span></span>
<span><span class="co">#&gt;   ENSG00000141556 101929597 -9.07984e-06   6.16583 -4.49543e-04    0.999642</span></span>
<span><span class="co">#&gt;   ENSG00000125804    728882 -2.02864e-05   2.75543 -2.93079e-04    0.999766</span></span>
<span><span class="co">#&gt;   ENSG00000228998      &lt;NA&gt;  1.05697e-05   4.14580  2.50417e-04    0.999800</span></span>
<span><span class="co">#&gt;   ENSG00000149636     79980  2.46976e-06   2.96401  1.02959e-04    0.999918</span></span>
<span><span class="co">#&gt;   ENSG00000122644     10124  1.62178e-06   4.51605  4.83868e-05    0.999961</span></span>
<span><span class="co">#&gt;                     adj.P.Val         B</span></span>
<span><span class="co">#&gt;                     &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000019169 0.000084371   3.65960</span></span>
<span><span class="co">#&gt;   ENSG00000260400 0.325280946   1.21448</span></span>
<span><span class="co">#&gt;   ENSG00000011201 0.325280946   1.55562</span></span>
<span><span class="co">#&gt;   ENSG00000068438 0.325280946   1.43237</span></span>
<span><span class="co">#&gt;   ENSG00000151229 0.327315930   1.15523</span></span>
<span><span class="co">#&gt;               ...         ...       ...</span></span>
<span><span class="co">#&gt;   ENSG00000141556    0.999863  -6.30298</span></span>
<span><span class="co">#&gt;   ENSG00000125804    0.999911  -5.80161</span></span>
<span><span class="co">#&gt;   ENSG00000228998    0.999911  -6.07834</span></span>
<span><span class="co">#&gt;   ENSG00000149636    0.999961  -5.84217</span></span>
<span><span class="co">#&gt;   ENSG00000122644    0.999961  -6.13228</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="co">## Optionally check the memory size</span></span>
<span><span class="co"># lobstr::obj_size(de_genes_prenatal_human_brain_smoking)</span></span>
<span><span class="co"># 3.73 MB</span></span>
<span></span>
<span><span class="co">## Access data of human genes as normally do with other GenomicRanges::GRanges()</span></span>
<span><span class="co">## objects or re-cast it as a data.frame</span></span>
<span><span class="va">de_genes_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">de_genes_prenatal_human_brain_smoking</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">de_genes_df</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 seqnames     start       end  width strand Length  Symbol</span></span>
<span><span class="co">#&gt; ENSG00000019169     chr2 119699742 119752236  52495      +   2079   MARCO</span></span>
<span><span class="co">#&gt; ENSG00000260400    chr10  70458257  70460551   2295      +   2295        </span></span>
<span><span class="co">#&gt; ENSG00000011201     chrX   8496915   8700227 203313      -   7131    KAL1</span></span>
<span><span class="co">#&gt; ENSG00000068438     chrX  48334541  48344752  10212      +   2740   FTSJ1</span></span>
<span><span class="co">#&gt; ENSG00000151229    chr12  40148823  40499891 351069      -  10027 SLC2A13</span></span>
<span><span class="co">#&gt; ENSG00000136319    chr14  20724717  20774153  49437      -   9705    TTC5</span></span>
<span><span class="co">#&gt;                 EntrezID       logFC   AveExpr         t      P.Value</span></span>
<span><span class="co">#&gt; ENSG00000019169     8685 -1.60327659 -1.801830 -6.145143 4.669893e-09</span></span>
<span><span class="co">#&gt; ENSG00000260400       NA  0.15158127  1.171418  4.098360 6.182984e-05</span></span>
<span><span class="co">#&gt; ENSG00000011201     3730  0.14231431  4.245759  4.093917 6.292772e-05</span></span>
<span><span class="co">#&gt; ENSG00000068438    24140 -0.04950860  4.306597 -4.059746 7.201659e-05</span></span>
<span><span class="co">#&gt; ENSG00000151229   114134  0.08427416  7.026246  4.001149 9.058392e-05</span></span>
<span><span class="co">#&gt; ENSG00000136319    91875 -0.09445164  3.683599 -3.931427 1.186235e-04</span></span>
<span><span class="co">#&gt;                    adj.P.Val        B</span></span>
<span><span class="co">#&gt; ENSG00000019169 8.437096e-05 3.659604</span></span>
<span><span class="co">#&gt; ENSG00000260400 3.252809e-01 1.214477</span></span>
<span><span class="co">#&gt; ENSG00000011201 3.252809e-01 1.555617</span></span>
<span><span class="co">#&gt; ENSG00000068438 3.252809e-01 1.432373</span></span>
<span><span class="co">#&gt; ENSG00000151229 3.273159e-01 1.155227</span></span>
<span><span class="co">#&gt; ENSG00000136319 3.497224e-01 1.000380</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.18/smokingMouse" class="external-link">smokingMouse</a></em>
package and the <a href="https://github.com/LieberInstitute/smokingMouse_Indirects" class="external-link">smoking
mouse</a> project were made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.18/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li><em><a href="https://bioconductor.org/packages/3.18/AnnotationHub" class="external-link">AnnotationHub</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=cowplot" class="external-link">cowplot</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/edgeR" class="external-link">edgeR</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.18/ExperimentHub" class="external-link">ExperimentHub</a></em>
(Morgan and Shepherd, 2023)</li>
<li><em><a href="https://CRAN.R-project.org/package=jaffelab" class="external-link">jaffelab</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=ggrepel" class="external-link">ggrepel</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/scater" class="external-link">scater</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=rlang" class="external-link">rlang</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=gridExtra" class="external-link">gridExtra</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/variancePartition" class="external-link">variancePartition</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/limma" class="external-link">limma</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=VennDiagram" class="external-link">VennDiagram</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=biomartr" class="external-link">biomartr</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=Hmisc" class="external-link">Hmisc</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=R.utils" class="external-link">R.utils</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/clusterProfiler" class="external-link">clusterProfiler</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/org.Mm.eg.db" class="external-link">org.Mm.eg.db</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=pheatmap" class="external-link">pheatmap</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/GenomicRanges" class="external-link">GenomicRanges</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.18/rtracklayer" class="external-link">rtracklayer</a></em></li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.18/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2023-07-21 22:01:28 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 12.926 secs</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-07-21</span></span>
<span><span class="co">#&gt;  pandoc   3.1.1 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi            1.63.2     <span style="color: #555555;">2023-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationHub          * 3.9.1      <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  backports                1.4.1      <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bibtex                   0.5.1      <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Biobase                * 2.61.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocFileCache          * 2.9.1      <span style="color: #555555;">2023-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics           * 0.47.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager              1.30.21.1  <span style="color: #555555;">2023-07-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  BiocStyle              * 2.29.1     <span style="color: #555555;">2023-07-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocVersion              3.18.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings               2.69.2     <span style="color: #555555;">2023-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bit                      4.0.5      <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bit64                    4.0.5      <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bitops                   1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  blob                     1.2.4      <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bookdown                 0.34       <span style="color: #555555;">2023-05-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bslib                    0.5.0      <span style="color: #555555;">2023-06-09</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cachem                   1.0.8      <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cli                      3.6.1      <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  crayon                   1.5.2      <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  curl                     5.0.1      <span style="color: #555555;">2023-06-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  DBI                      1.1.3      <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  dbplyr                 * 2.3.3      <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  DelayedArray             0.27.9     <span style="color: #555555;">2023-07-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                     1.4.2      <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  digest                   0.6.33     <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  dplyr                    1.1.2      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ellipsis                 0.3.2      <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  evaluate                 0.21       <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ExperimentHub          * 2.9.1      <span style="color: #555555;">2023-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  fansi                    1.0.4      <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fastmap                  1.1.1      <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  filelock                 1.0.2      <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fs                       1.6.3      <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  generics                 0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb           * 1.37.2     <span style="color: #555555;">2023-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData         1.2.10     <span style="color: #555555;">2023-07-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges          * 1.53.1     <span style="color: #555555;">2023-05-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  glue                     1.6.2      <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  htmltools                0.5.5      <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  httpuv                   1.6.11     <span style="color: #555555;">2023-05-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  httr                     1.4.6      <span style="color: #555555;">2023-05-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  interactiveDisplayBase   1.39.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  IRanges                * 2.35.2     <span style="color: #555555;">2023-06-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  jquerylib                0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  jsonlite                 1.8.7      <span style="color: #555555;">2023-06-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  KEGGREST                 1.41.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  knitr                    1.43       <span style="color: #555555;">2023-05-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  later                    1.3.1      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lattice                  0.21-8     <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lifecycle                1.0.3      <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lubridate                1.9.2      <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  magrittr                 2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Matrix                   1.6-0      <span style="color: #555555;">2023-07-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics         * 1.13.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats            * 1.0.0      <span style="color: #555555;">2023-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  memoise                  2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  mime                     0.12       <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pillar                   1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pkgdown                  <span style="color: #BB00BB; font-weight: bold;">2.0.7.9000</span> <span style="color: #555555;">2023-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (r-lib/pkgdown@c920680)</span></span></span>
<span><span class="co">#&gt;  plyr                     1.8.8      <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  png                      0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  promises                 1.2.0.1    <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  purrr                    1.0.1      <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  R6                       2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ragg                     1.2.5      <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rappdirs                 0.3.3      <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Rcpp                     1.0.11     <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RCurl                    1.98-1.12  <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RefManageR             * 1.4.0      <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rlang                    1.1.1      <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                2.23       <span style="color: #555555;">2023-07-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rprojroot                2.0.3      <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RSQLite                  2.3.1      <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  S4Arrays                 1.1.4      <span style="color: #555555;">2023-05-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  S4Vectors              * 0.39.1     <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                     0.4.7      <span style="color: #555555;">2023-07-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  sessioninfo            * 1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  shiny                    1.7.4.1    <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  smokingMouse           * 0.99.9     <span style="color: #555555;">2023-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  SparseArray              1.1.10     <span style="color: #555555;">2023-06-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  stringi                  1.7.12     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  stringr                  1.5.0      <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment   * 1.31.1     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  systemfonts              1.0.4      <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  textshaping              0.3.6      <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tibble                   3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidyselect               1.2.0      <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  timechange               0.2.0      <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  utf8                     1.2.3      <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  vctrs                    0.6.3      <span style="color: #555555;">2023-06-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  withr                    2.5.0      <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xfun                     0.39       <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xml2                     1.3.5      <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xtable                   1.8-4      <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  XVector                  0.41.1     <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                     2.3.7      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                 1.47.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.18/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.23. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[2]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2023experimenthub"></a><a href="#cite-morgan2023experimenthub">[3]</a><cite>
M. Morgan and L. Shepherd. <em>ExperimentHub: Client to access
ExperimentHub resources</em>. R package version 2.9.1. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.ExperimentHub" class="external-link">10.18129/B9.bioc.ExperimentHub</a>.
URL:
<a href="https://bioconductor.org/packages/ExperimentHub" class="external-link">https://bioconductor.org/packages/ExperimentHub</a>.</cite>
</p>
<p>
<a id="bib-ole2023biocstyle"></a><a href="#cite-ole2023biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.29.1. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2023language"></a><a href="#cite-2023language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2023. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2. 2021. URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2023knitr"></a><a href="#cite-xie2023knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.43. 2023. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Daianna Gonzalez-Padilla.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
